parserator_web/views.py

```diff
@@ -14,11 +15,48 @@ class AddressParse(APIView):
     renderer_classes = [JSONRenderer]
 
     def get(self, request):
-        # TODO: Flesh out this method to parse an address string using the
-        # parse() method and return the parsed components to the frontend.
-        return Response({})
+        """Returns the string the user sent, the parsed address
+        components and the address type if the string is a valid address"""
+
+        # Get the unparsed address from JSON object query
+        input_string = request.query_params['address']
+
+        try:
+
+            # Get address components using .parse() method
+            address_components, address_type = self.parse(input_string)
+
+            # Return a dictionary object
+            return Response({
+                'input_string': input_string,
+                'address_components': address_components,
+                'address_type': address_type
+            })
+
+        except Exception as e:
+
+            # Return the expected dictionary object, including error info
+            # and setting HTTP status to 400
+
+            return self.handle_exception(e, input_string)
 
     def parse(self, address):
-        # TODO: Implement this method to return the parsed components of a
-        # given address using usaddress: https://github.com/datamade/usaddress
+        """Returns parsed components and address type of a given address
+        using usaddress"""
+
+        # Use .tag() from usaddress to parse address
+        address_components, address_type = usaddress.tag(address)
+
+        # Return address_components as OrderedDict and address_type as Str
         return address_components, address_type
+
+    def handle_exception(self, exc, input_string):
+        """Handles ParseError exceptions stemming from invalid address
+        type"""
+
+        return Response({
+                'input_string': input_string,
```

Comment:
If including the original data (great call!) in an API response, I like to return it under the same parameter it was sent, in this case "address".